binnacle/lib/services/weather_service.dart

```diff
@@ -7,45 +7,84 @@ import 'package:rxdart/rxdart.dart';
 import '../models/wind_model.dart';
-import '../providers/wind_provider.dart';
+import '../models/wind_service_interface.dart';
+import '../models/service_data.dart';
+import './service_wrapper_interface.dart';
 
 class WeatherService extends IWindService {
   Client client = Client();
   final _apiKey = "80823ccc590c29c76f3094869dcdbee9";
   final _apiURL = "https://api.openweathermap.org/data/2.5/weather";
   var _windStream = StreamController<WindModel>();
   BehaviorSubject<PositionModel> _position;
-
+  final ServiceData serviceData = ServiceData('wind', 'name', 1);
 
   WeatherService(BehaviorSubject<PositionModel> position) {
     this._position = position;
-    fetchWeather(_position.value);
+
+    fetchWeather(_position);
   }
+  Future<PositionModel> lastNonNull(Stream<PositionModel> stream) =>
+      stream.firstWhere((x) => x != null);
 
-  void fetchWeather(PositionModel position) async {
+  void fetchWeather(BehaviorSubject<PositionModel> positionStream) async {
+    var position;
+    try {
+      position = await lastNonNull(positionStream);
+    } catch (e) {
+      print(e);
+      print(
+          "From weather_service: If you aren't in the ios simulator, location is messing up and threw this ^^^");
+      position = new PositionModel(lat: 0.0, lon: 0.0, speed: 0.0);
+    }
+    print("___ pos const: " + position.lat.toString());
     print("fetching weather....");
     print(position.lat.toString() + ' ' + position.lon.toString());
-    final response = await client
-        .get((_apiURL 
-          + "?lat=" + position.lat.toString()
-          + "&lon=" + position.lon.toString() 
-          + "&APPID="+_apiKey
-        ));
-
-    print(response.body.toString());
+    final response = await client.get((_apiURL +
+        "?lat=" +
+        position.lat.toString() +
+        "&lon=" +
+        position.lon.toString() +
+        "&APPID=" +
+        _apiKey));
+
     if (response.statusCode == 200) {
+      // TODO remove this debugging code
       // If the call to the server was successful, parse the JSON
       var temp = WeatherModel.fromJson(json.decode(response.body));
       WindModel wind = WindModel(temp.wind.speed, temp.wind.deg);
-      print("Temp:" + temp.toString());
-      print(temp.wind.speed.toString() + "   " +  temp.wind.deg.toString());
       _windStream.sink.add(wind);
     } else {
       // If that call was not successful, throw an error.
-      throw Exception('Failed to load weather');
+      // !TODO  Don't throw the error. Add the error into the stream
+      print("RESPONSE status code: " +
+          response.statusCode.toString() +
+          " RESPONSE BODY:  " +
+          response.body.toString());
+      //throw Exception('Failed to load weather');
     }
   }
 
+    dispose() async {
+    //! there's no subscibtion here like there are in other services
+    //! consider starting debug here if issue arises
+    // await _subscription.pause(); 
+    await _windStream.close();
+
+  }
```

Comment:
Good question. I meant for Will to add it, because I didn't understand what he did. I mentioned that at a scrum meeting, but I forgot to make it happen. That's my bad